src/platform/rp2040/rp2040Watchdog.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+#include "configuration.h"
+
+class Rp2040Watchdog : public concurrency::OSThread
+{
+  public:
+    explicit Rp2040Watchdog() : OSThread("Rp2040Watchdog")
+    {
+        LOG_TRACE("Rp2040Watchdog::Rp2040Watchdog(): Initializing\n");
+        watchdogIsRunning = false;
+
+        // Todo: We need a port of the 'Preferences' type to rp2040 so that
+        //       we can keep track of reboots
+    }
+
+    void reset() { lastResetTick = to_ms_since_boot(get_absolute_time()); }
+
+  protected:
+    int32_t runOnce() override
+    {
+        // If we are long enough into boot/initialization, start the hardware watchdog,
+        // otherwise check if timer is below timeout and update the watchdog.
+        uint32_t currentTick = to_ms_since_boot(get_absolute_time());
+        if (!watchdogIsRunning) {
+
+            LOG_TRACE("Rp2040Watchdog->runOnce(): watchdog not running (uptime = %u seconds)\n", currentTick / 1000);
+
+            if (currentTick > 30 * 1000) { // 30 seconds into boot
+                LOG_INFO("Rp2040Watchdog->runOnce(): starting watchdog (currentTick = %u seconds)\n", currentTick / 1000);
+                watchdog_enable(0x7fffff, true);
+                watchdogIsRunning = true;
+                lastResetTick = currentTick;
+            }
+        } else {
+
+            // Check for rollover, each 49 days. If so, then assume all is good
+            if (currentTick < lastResetTick) {
+
+                LOG_TRACE("Rp2040Watchdog->runOnce(): rollover detected (lastResetTick = %lu minutes)\n",
+                          lastResetTick / (60 * 1000));
+                lastResetTick = currentTick;
+            }
+
+            // Time since last reset
+            uint32_t timeout = currentTick - lastResetTick;
+
+            // Update watchdog if timeout is below 90 seconds, same as esp32 watchdog
+            if (timeout < 90 * 1000) {
+                watchdog_update();
```

Comment:
This is not going to work, right? In the worst case you only call `watchdog_update()` just before the 90 seconds expires, but it already triggers after 8 seconds.

Reply:
timeout needs to be below 90 seconds, in most cases its close to 0 (zero) (due to soft-updates from the loop or other places), so it will be called aprox. each. 4. second - but I will add a trace line so that we can verify this. np.

Reply:
I mean I don't see why you're checking for less than 90 seconds? It will never be more than 8 seconds, because then it reboots already.